node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -559,20 +559,23 @@ object P2PClient extends P2PLogger {
           config)
 
   def apply(
-      context: ActorRefFactory,
       peer: Peer,
       peerMessageReceiver: PeerMessageReceiver,
       onReconnect: Peer => Future[Unit],
       onStop: Peer => Future[Unit],
-      maxReconnectionTries: Int = 16)(implicit
-      config: NodeAppConfig): P2PClient = {
-    val actorRef = context.actorOf(props = props(peer,
-                                                 peerMessageReceiver,
-                                                 onReconnect,
-                                                 onStop,
-                                                 maxReconnectionTries),
-                                   name =
-                                     BitcoinSNodeUtil.createActorName(getClass))
+      maxReconnectionTries: Int = 16,
+      supervisor: ActorRef)(implicit config: NodeAppConfig): P2PClient = {
+
+    val clientProps = props(peer,
+                            peerMessageReceiver,
+                            onReconnect,
+                            onStop,
+                            maxReconnectionTries)
+
+    implicit val timeout: Timeout = Timeout(10.second)
+    val actorRefF = supervisor ? clientProps
+    val actorRef =
```

Comment:
Lets avoid blocking here, can we just return a `Future[P2PClient]` from this method?